src/commands/create.js

```diff
@@ -76,153 +48,121 @@ export default asyncCommand({
+			}
 
-		let exists = false;
-		try {
-			exists = (await fs.stat(target)).isDirectory();
-		}
-		catch (err) {}
-
-		if (exists && argv.force) {
-			const question = {
-				type: 'confirm',
-				name: 'enableForce',
-				message: `You are using '--force'. Do you wish to continue?`,
-				default: false,
-			};
-
-			let { enableForce } = await inquirer.prompt(question);
-
-			if (enableForce) {
-				process.stdout.write('Initializing project in the current directory...\n');
-			} else {
-				process.stderr.write(chalk.red('Error: Cannot initialize in the current directory\n'));
-				process.exit(1);
+			if (exists && argv.force) {
+				let { enableForce } = await prompt({
+					type: 'confirm',
+					name: 'enableForce',
+					message: `You are using '--force'. Do you wish to continue?`,
+					default: false
+				});
+
+				if (enableForce) {
+					process.stdout.write('Initializing project in the current directory...\n');
+				} else {
+					return error('Refusing to overwrite current directory!', 1);
+				}
 			}
+		} else {
+			// TODO: interactive
 		}
 
-		if (exists && !argv.force) {
-			process.stderr.write(chalk.red('Error: Cannot initialize in the current directory, please specify a different destination or use --force\n'));
-			process.exit(1);
-		}
+		let repo = TEMPLATES[argv.template] || argv.template;
+
+		// Attempt to fetch the `template`
+		let archive = await gittar.fetch(repo).catch(err => {
+			err = err || { message:'An error occured while fetching template.' };
+			return error(err.code === 404 ? `Could not find repostory: ${repo}` : err.message, 1);
+		});
 
 		let spinner = ora({
 			text: 'Creating project',
 			color: 'magenta'
 		}).start();
 
-		if (!exists) {
-			await Promise.promisify(mkdirp)(target);
-		}
-
-		await copy(
-			path.resolve(__dirname, '../..', template),
-			target,
-			{ filter: ['**/*', '!build'] }
-		);
-
-		spinner.text = 'Initializing project';
-
-		await initialize(argv.yarn, target);
+		// Extract files from `archive` to `target`
+		await gittar.extract(archive, target, {
+			strip: 2,
+			filter(path) {
+				// TODO: remove `/build/` ??
+				// TODO: read & respond to meta/hooks
+				return path.includes('/template/') && !path.includes('/build/');
+			}
+		});
 
-		let pkg = JSON.parse(await fs.readFile(path.resolve(target, 'package.json')));
+		spinner.text = 'Parsing `package.json` file';
 
-		pkg.scripts = await pkgScripts(argv.yarn, pkg);
+		// Validate user's `package.json` file
+		let pkgData;
+		let pkgFile = resolve(target, 'package.json');
 
-		try {
-			await fs.stat(path.resolve(target, 'src'));
-		}
-		catch (err) {
-			pkg.scripts.test = pkg.scripts.test.replace('src', '.');
+		if (pkgFile) {
+			pkgData = JSON.parse(await fs.readFile(pkgFile));
+			// Write default "scripts" if none found
+			pkgData.scripts = pkgData.scripts || (await addScripts(pkgData, target, isYarn));
```

Comment:
> we could merge our scripts into the template's?

I don't think there is a necessary for that since templates will have scripts defined in the package.json file (includes our scripts too)

Reply:
Just sucks if the cli wants to change script down the line 😋